### DISCUSSION TITLE
Guidance on pagination and bulk selection for Data Grid

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1943] [Q&A] [answered]
To start off, just want to say that the Data Grid component looks great!!

I have a query around the ideal behaviour when using `select all` on a table/grid with pagination. 

I can see in your [codesandbox prototype](https://pvfs9.csb.app/) that `select all` selects all rows on all pages of the Data Grid.

I'm not entirely sure about selecting rows that the user can't see. Is this customisable so that only the current page is selected? And we can add something similar to what Gmail does as shown below: 

![image](https://user-images.githubusercontent.com/72031986/136974798-783538bd-581b-4346-ab80-e82d2b06b80d.png)

**Comment by SiTaggart** [comment, marked as answer]
Hi @PoojMenon 

Everything about the data handling with Data Grid is essentially up to the application builder. So you can do what ever you would like to do with it, that's kind of the beauty of its flexibility.

You should be able to achieve something very similar to Gmail, but your product engineers will have to handle it in the application state itself. Data grid just renders what you tell it to render.

Looking forward to seeing it's first use. 🎉

**Reply by PoojMenon** [reply]
Thanks for that explanation - that's exactly what I was hoping for! Thank you!
